pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -593,12 +594,15 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 			}
 			mc.ObjectMeta.UID = uuid.NewUUID()
 			_, ok := cfg.GetAnnotations()[ctrlcommon.MCNameSuffixAnnotationKey]
-			arr := strings.Split(managedKey, "-")
 			// If the MC name suffix annotation does not exist and the managed key value returned has a suffix, then add the MC name
 			// suffix annotation and suffix value to the kubelet config object
-			if len(arr) > 4 && !ok {
-				if err := ctrl.addAnnotation(cfg, ctrlcommon.MCNameSuffixAnnotationKey, arr[len(arr)-1]); err != nil {
-					return ctrl.syncStatusOnly(cfg, err, "could not update annotation for kubeletConfig")
+			if strings.Contains(managedKey, "-generated-kubelet-") && !ok {
```

Comment:
We shouldn't need this check as the managed key will always have `generated-kubelet` in it as this controller is for the kubeletconfig stuff only.